ember-amount-input/src/components/amount-input.ts

```diff
@@ -0,0 +1,180 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import './amount-input.css';
+
+const KEY_CODE_E = 69;
+const KEY_CODE_FULLSTOP = 190;
+const KEY_CODE_COMMA = 188;
+
+export interface AmountInputArgs {
+  /**
+   * The currency displayed in the input
+   */
+  currency?: string;
+
+  /**
+   * Disables the input
+   */
+  disabled?: boolean;
+
+  /**
+   * A custom class applied on the input
+   */
+  inputClass?: string;
+
+  /**
+   * A custom id applied on the input
+   */
+  inputId?: string;
+
+  /**
+   * Specifies the minimum value for the input field
+   */
+  min?: number;
+
+  /**
+   * Specifies the maximum value for the input field
+   */
+  max?: number;
+
+  /**
+   * Specifies the number of decimals to use for the amount value
+   */
+  numberOfDecimal?: number;
+
+  /**
+   * The placeholder displayed in the input
+   */
+  placeholder?: string;
+
+  /**
+   * Specifies if the input field should be read-only
+   */
+  readonly?: boolean;
+
+  /**
+   * Specifies the number intervals for the input field
+   */
+  step?: number;
+
+  /**
+   * The callback function triggered when the input value is updated
+   */
+  update: (value: string) => void;
+
+  /**
+   * The value of the input. It should be updated using the `update` argument
+   */
+  value: number | string;
```

Comment:
This one was a tricky (and not perfect) choice, so happy to discuss it.

What's the reason behind this choice? Our official documentation refers to a `number` type for the `value` argument. However, we can support both `string` and `number`. Using a `number` seems more natural, but the documentation also mentions that the `value` should be updated using the `update` function argument, which only returns a `string`.

Reply:
To better understand this: In which case do we need to accept a `string` as `value`?

Couldn't we make the `update` function accept a `number` instead of `string`? Maybe that will solve this issue? 

From what I see also `min` and `max` accept a `number`, so it feels weird that we're accepting a `string` as `value` here, what do you think?

Reply:
> To better understand this: In which case do we need to accept a string as value?

I'd say in case a user wants to keep everything type synced. The addon usage requires passing a value to the input component, and updating this value using a callback for the update argument, which only returns a string. If, as a user, I don't want to have to parse this string before updating my value, then it can be better to accept string for the value as well.

> Couldn't we make the update function accept a number instead of string? Maybe that will solve this issue?

It will probably solve it, but this would be quite a breaking change, right? Don't we want to avoid this for now?

> From what I see also min and max accept a number, so it feels weird that we're accepting a string as value here, what do you think?

I agree yes. For min and max we have no choice, because if we use string for them, the browser will not interpret them as numerical values, and the input may not behave as expected.

Reply:
> It will probably solve it, but this would be quite a breaking change, right? Don't we want to avoid this for now?

Yeah, I'd avoid doing this right now because indeed it would be a breaking change. Could you log an issue about this? 🙏  So that we can tackle this after the TypeScript migration.

Reply:
The [issue](https://github.com/qonto/ember-amount-input/issues/613) was created. Feel free to edit if you feel like something is missing 👍

Reply:
LGTM 🙌